include/deal.II/grid/grid_tools.h

```diff
@@ -1438,7 +1515,31 @@ namespace GridTools
         }
   }
 
-
+  template <class Container, typename Predicate>
+  void
+  compute_active_cell_halo_layer (const Container                                       &container,
+                                  const Predicate                                       &predicate,
+                                  std::vector<typename Container::active_cell_iterator> &active_halo_layer)
+  {
+    active_halo_layer.clear ();
+    std::vector<bool> locally_active_vertices_on_subdomain (get_tria(container).n_vertices(),
```

Comment:
I did not steal this idea from `mark_locally_active_vertices_on_level()` in `dof_handler_policy.cc`. No sir, not me and not from there...